File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L2/05/arraylebihdarisatu.kt
package IR

import java.util.*

object arraylebihdarisatu {
    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)


        print("Enter a 4 by 4 matrix row by row: ")
        val matrix = Array(4) { DoubleArray(4) } //inisialisai array 2 dimensi

        for (x in 0..3) {
            for (y in 0..3) {
                matrix[x][y] = sc.nextDouble() //menginput angka ke dalam array 2d
            }
        }

        print("Sum of the elements in the major  diagonal is  " + HitungDiagonal(matrix)) //menampilkan fungsi hasil perhitungan diagonal
    }

    fun HitungDiagonal(matriks: Array<DoubleArray>): Double { //fungsi untuk mengitung diagonal
        var total = 0.0
        for (i in matriks.indices) {
            total += matriks[i][i]
        }
        return total
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L4/06/arraylebihdarisatu.ts
import { Scanner } from 'readline-sync';

function main(): void {
  const sc = new Scanner();

  console.log('Enter a 4 by 4 matrix row by row: ');
  const matrix: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));

  for (let x = 0; x < 4; x++) {
    for (let y = 0; y < 4; y++) {
      matrix[x][y] = sc.nextDouble();
    }
  }

  let hasil: number = 0;
  for (let i = 0; i < matrix.length; i++) {
    hasil += matrix[i][i];
  }

  console.log(`Sum of the elements in the major diagonal is ${hasil}`);
}

main();

